api/v1alpha1/job_name_test.go

```diff
@@ -0,0 +1,25 @@
+package v1alpha1_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vshn/k8up/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestJobName(t *testing.T) {
+	name := "test-for-job-name"
+	subjects := []v1alpha1.JobObject{
+		&v1alpha1.Archive{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		&v1alpha1.Backup{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		&v1alpha1.Check{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		&v1alpha1.Prune{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		&v1alpha1.Restore{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		&v1alpha1.Schedule{ObjectMeta: metav1.ObjectMeta{Name: name}},
+	}
+
+	for _, subject := range subjects {
+		assert.Equal(t, subject.GetType().String()+"-"+name, subject.GetJobName())
```

Comment:
I would prefer to not compute the expected value but have it hard coded. For that I would make `subjects` a dictionary containing both the expected job name and the subject (`map[string]v1alpha1.JobObject`). While at it, I also would use nested test functions. This makes it esier to identify the exact case when there is a test failure.